deepmd/pt/model/descriptor/descriptor.py

```diff
@@ -56,15 +60,127 @@ class SomeDescript(Descriptor):
         return Descriptor.__plugins.register(key)
 
     @classmethod
-    def get_stat_name(cls, config):
+    def get_stat_name(cls, config, ntypes):
+        """
+        Get the name for the statistic file of the descriptor.
+        Usually use the combination of descriptor name, rcut, rcut_smth and sel as the statistic file name.
+        """
+        if cls is not Descriptor:
+            raise NotImplementedError("get_stat_name is not implemented!")
         descrpt_type = config["type"]
-        return Descriptor.__plugins.plugins[descrpt_type].get_stat_name(config)
+        return Descriptor.__plugins.plugins[descrpt_type].get_stat_name(config, ntypes)
 
     @classmethod
     def get_data_process_key(cls, config):
```

Comment:
very bad design of passing-in a dict. explicitly passing the parameters is preferred.